File: seeker/snippet/airwrk.go
//date: 2023-01-09T17:07:22Z
//url: https://api.github.com/gists/3acb34911c2d9eb6863be09eaef5f0c7
//owner: https://api.github.com/users/diptomondal007

// Problem 1
func isPalindrome(num int) bool {
	s := strconv.FormatInt(int64(num), 10)
	if s[0] == s[len(s)-1] {
		return true
	}
	return false
}

// Problem 2
func singleNumber(arr []int) int {
	m := make(map[int]int)
	for i := range arr {
		m[arr[i]]++
	}

	for k := range m {
		if m[k] == 1 {
			return k
		}
	}
	return 0
}

// Problem 3
func goodStringLen(words []string, chars string) int {
	m := make(map[uint8]int)
	for i := range chars {
		m[chars[i]]++
	}

	var length int
	for i := range words {
		valid := true
		tempMap := make(map[uint8]int)
		for j := range words[i] {
			val := m[words[i][j]]
			//log.Println(string(words[i][j]), val)
			//cat
			if tempMap[words[i][j]] > val || val < 1 {
				//log.Println(string(words[i][j]), tempMap[words[i][j]])
				valid = false
				break
			}
			tempMap[words[i][j]]++
		}
		if valid {
			//log.Println(words[i])
			length += len(words[i])
		}
	}

	return length
}

File: seeker/snippet/airwrk.py
#date: 2024-08-27T16:59:54Z
#url: https://api.github.com/gists/aa376db8b841b1d90d9e0c56f6509272
#owner: https://api.github.com/users/ahmadalsajid

# Problem 1
def div_sum(n):
    return sum([int(i) for i in list(str(n))])

def solution1(num):
    while num > 9:
        num = div_sum(num)
    print(num)

solution1(num = 38)


# Problem 2
def solution2(nums):
    _set = set(nums)
    if len(_set) == len(nums):
        print('false')
    else:
        print('true')

solution2(nums = [1,1,1,3,3,4,3,2,4,2])


# Problem 3
def solution3(s):
    vowels = 'aeiouAEIOU'
    _temp_vowels = list()
    _temp_string = list()

    for letter in s:
        if letter in vowels:
            _temp_vowels.append(letter)
            _temp_string.append('_')
        else:
            _temp_string.append(letter)

    _temp_vowels.reverse()
    _index = 0
    for _indx, letter in enumerate(_temp_string):
        if letter == '_':
            _temp_string[_indx] = _temp_vowels[_index]
            _index = _index + 1

    print(''.join(_temp_string))

solution3(s = 'algorithm')
